### PR TITLE
Add support for int and long fields in count_by expression

### PR DESCRIPTION
**Pull request by raronson**
@markhibberd @charleso

### CODE REVIEW

File: ivory-cli/src/test/bin/tests/window/expected/dictionary.psv [hunk 1]
```diff
@@ -1,58 +1,62 @@
...
+52|halibut_state|minimum_in_weeks|int|||NA
+53|halibut_state|quantile_in_days|double|||NA
+54|halibut_state|quantile_in_weeks|double|||NA
+55|halibut_state|std_dev|double|||NA
+56|halibut_state|sum|double|||NA
+57|halibut_state|swiss_cheese|double|||NA
+58|mackerel|days_catch|[string]|||NA
+59|mackerel|mackerel|(common_name:string,size:double)|||NA
+60|mackerel|size_by_common_name|()|||NA
+61|mackerel|unique_catches|[string]|||NA
```
Comment (markhibberd) [inline ivory-cli/src/test/bin/tests/window/expected/dictionary.psv]: Would be good if we could have a count_by,<field> test on a struct.

**Comment by markhibberd** [comment]
:+1: Looks good after extra test.

**Comment by markhibberd** [comment]
:shaved_ice:

**Comment by charleso** [comment]
:baby:

**Pull request merged**